plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -4,93 +4,112 @@ import (
 	_ "github.com/denisenkom/go-mssqldb" // go-mssqldb initialization
 )
 
+//------------------------------------------------------------------------------------------------
+//------------------ Azure SQL Database ------------------------------------------------------
+//------------------------------------------------------------------------------------------------
 // Only executed if AzureDB flag is set
-const sqlAzureDBResourceStats string = `SET DEADLOCK_PRIORITY -10;
-IF SERVERPROPERTY('EngineEdition') = 5  -- Is this Azure SQL DB?
-BEGIN
-        SELECT TOP(1)
-                'sqlserver_azure_db_resource_stats' AS [measurement],
-                REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
-                DB_NAME() as [database_name],
-                cast(avg_cpu_percent as float) as avg_cpu_percent,
-                cast(avg_data_io_percent as float) as avg_data_io_percent,
-                cast(avg_log_write_percent as float) as avg_log_write_percent,
-                cast(avg_memory_usage_percent as float) as avg_memory_usage_percent,
-                cast(xtp_storage_percent as float) as xtp_storage_percent,
-                cast(max_worker_percent as float) as max_worker_percent,
-                cast(max_session_percent as float) as max_session_percent,
-                dtu_limit,
-                cast(avg_login_rate_percent as float) as avg_login_rate_percent  ,
-                end_time,
-                cast(avg_instance_memory_percent as float) as avg_instance_memory_percent ,
-                cast(avg_instance_cpu_percent as float) as avg_instance_cpu_percent
-        FROM
-                sys.dm_db_resource_stats WITH (NOLOCK)
-        ORDER BY
-                end_time DESC
+const sqlAzureDBResourceStats string = `
+SET DEADLOCK_PRIORITY -10;
+IF SERVERPROPERTY('EngineEdition') <> 5 BEGIN /*not Azure SQL DB*/
+	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - the instance "'+ @@SERVERNAME +'" is not an Azure SQL DB. Check the database_type parameter in the telegraf configuration.';
```

Comment:
Can we change the message thrown to include DB name? on Azure DB there are many SQL Databases.. in particular... this is an example of the MI message -- giving just a logical server doesn't help customer identify which DB.. This is accross the board I think in Azure and on-prem
	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - Connection string Server:"'+ @@SERVERNAME + ',Database:' + DB_NAME() +'" is not an Azure Managed Instance. Check the database_type parameter in the telegraf configuration.';

Reply:
Makes sense for Azure DB since the DB is part of the identifier, I'm not sure about Managed instance, but for the on-prem that's not really useful since DB_NAME() will probably just return "master" all the time as database name.

I'll add it in the Azure section as you asked

Reply:
You need it everywhere.. think of when you put an Azure DB connection string in say an on-prem section or a managed instance section

Reply:
I've edited the error message to include the Database name.